### PR TITLE
Update to treatment of vertical passive tracers and a new pseudo-salt tracer

### PR DESCRIPTION
**Pull request by ashao**
The changes made here ensure that passive tracers are treated in the same way as the active tracers.
Overview of changes
-  tracer_vertdiff was moved from MOM_tracer_registry.F90 to MOM_tracer_diabatic.F90
-  The new subroutine applyTracerBoundaryFluxesInOut is an equivalent routine to applyBoundaryFluxesInOut. This is called during tracer_vertdiff if the appropriate options are passed into that routine.
-  This new subroutine also now allows passive tracers to be conserved when ALE is used
-  A new passive tracer, dubbed 'pseudo_salt' has been added that uses the same initial conditions and boundary fluxes as the thermodynamic salt tracer. The diagnostic 'pseudo_salt_diff' should be zero everywhere if active and passive tracers are treated the same.
-  A new run time flag has been added called 'TRACER_TRIDIAG' which uses tracer_vertdiff instead of tridiagTS to do vertical diffusion of T and S.
-  The fidelity of these changes was confirmed in a version of the Baltic test case configured for use in ALE z
-  The changes in this pull request should NOT change answers except for passive tracers

### CODE REVIEW

**Comment by ashao** [comment]
Also note that this does NOT fix the passive tracer concentration for tracer which use MOM_generic_tracer. generic_tracer_vertdiff_G must be updated similarly to tracer_vertdiff in MOM_tracer_diabatic.F90.

**Comment by ashao** [comment]
To validate this fix for the generic tracer frammework, I used a OM4 0.5 degree configuration based on John Krasting's CFC/SF6 model run. I integrated it for a day, then used it as a restart for a short 5-day run where surface fluxes of CFCs were turned off. Total CFC-11 and CFC-12 stock were conserved to the last 4 decimal places which is the same level of accuracy as passive tracers using the MOM6 tracer framework.

**Comment by jkrasting** [comment]
@ashao -

@adcroft, @nikizadehgfdl, and I were looking at this update.  Are any changes needed to ocean_shared for this to work?

**Comment by ashao** [comment]
@nikizadehgfdl , @adcroft , @jkrasting , 
No there should be no changes necessary to any of the ocean_shared routines. 

Originally, I thought there might need to be, but I was able to organize it so that the fix is only implemented on the MOM6 side via MOM_generic_tracer.F90.

Andrew

**Comment by jkrasting** [comment]
Excellent.  Thanks @ashao

-----
Dr. John Krasting, Physical Scientist
NOAA/Geophysical Fluid Dynamics Laboratory
Biogeochemistry, Ecosystems, and Climate Group
201 Forrestal Road, Princeton, NJ 08540
P. (609) 452-5359   |   F. (609) 987-5063

**Comment by jkrasting** [comment]
The fix from @ashao undoes the edits required for the generic_abiotic routines from commit 3b8ecaaa.  Thus, today's tag dev/master/2016.09.22 no longer works with the latest ocean_shared code.  I will put back in the necessary changes, but once it is done, can we retest and update the 2016.09.22 tag?

File: src/tracer/MOM_generic_tracer.F90 [hunk 1]
```diff
@@ -61,13 +61,12 @@ module MOM_generic_tracer
   use MOM_hor_index, only : hor_index_type
   use MOM_io, only : file_exists, read_data, slasher, vardesc, var_desc
   use MOM_restart, only : register_restart_field, query_initialized, MOM_restart_CS
-  use MOM_spatial_means, only : global_area_mean
```
Comment (adcroft) [inline src/tracer/MOM_generic_tracer.F90]: @jkrasting is right, this commit did undo the changes from https://github.com/NOAA-GFDL/MOM6/pull/351/files . I'm guessing this happened because he was merging in changes from a file that is being copied in via the XML and isn't even on GitHub.

**Comment by adcroft** [comment]
This is fixed now. Thanks @ashao and @jkrasting .
